<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <link href='https://fonts.googleapis.com/css?family=Chivo:900' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-dark.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <title>Bio-ontologies and their role in analyzing personal genome data by bio-ontology-research-group</title>
  </head>

  <body>
    <div id="container">
      <div class="inner">

        <header>
          <h1>Bio-ontologies and their role in analyzing personal genome data</h1>
          <h2>Information for the tutorial on ontologies and phenotypes</h2>
        </header>

        <section id="downloads" class="clearfix">
          <a href="https://github.com/bio-ontology-research-group/ontology-and-phenotype-tutorial/zipball/master" id="download-zip" class="button"><span>Download .zip</span></a>
          <a href="https://github.com/bio-ontology-research-group/ontology-and-phenotype-tutorial/tarball/master" id="download-tar-gz" class="button"><span>Download .tar.gz</span></a>
          <a href="https://github.com/bio-ontology-research-group/ontology-and-phenotype-tutorial" id="view-on-github" class="button"><span>View on GitHub</span></a>
        </section>

        <hr>

        <section id="main_content">
          <p>Welcome to the tutorial website for the tutorial "Bio-ontologies and their role in analyzing personal genome data". The tutorial will initially be presented at <a href="http://www.eccb2016.org/">ECCB 2016</a> on 4 September 2016.</p>

<h1>
<a id="aim" class="anchor" href="#aim" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Aim</h1>

<p>The aim of the tutorial is to introduce recent advances in <strong>using biomedical ontologies for data analysis</strong>, in particular for the <strong>interpretation of personal genome data</strong> and identifying <strong>causal variants</strong> underlying a phenotype or disease.</p>

<h1>
<a id="target-audience" class="anchor" href="#target-audience" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Target audience</h1>

<p>The tutorial is aimed at <strong>computational biologists</strong> and (bio-)medical <strong>informaticians</strong> interested in incorporating ontologies, in particular phenotype ontologies, into their analysis pipelines. <strong>Clinicians</strong>, <strong>biologists</strong>, and <strong>pharmacologists</strong> with basic informatics expertise will also find the course accessible and useful.</p>

<h1>
<a id="presenters" class="anchor" href="#presenters" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Presenters</h1>

<ul>
<li>
<a href="http://www.pdn.cam.ac.uk/directory/paul-schofield">Dr Paul N Schofield</a>, University Reader in Biomedical Informatics, Department of Physiology, Development and Neuroscience, University of Cambridge</li>
<li>
<a href="http://leechuck.de">Prof Robert Hoehndorf</a>, Assistant Professor of Computer Science, Computational Bioscience Research Center, King Abdullah University of Science and Technology</li>
<li>
<a href="http://www.birmingham.ac.uk/staff/profiles/cancer-genomic/gkoutos-georgios.aspx">Prof Georgios V Gkoutos</a>, Professor of Clinical Bioinformatics, College of Medical and Dental Sciences, Institute of Cancer and Genomic Sciences, Centre for Computational Biology, University of Birmingham</li>
<li>Ms Imene Boudellioua, PhD student, Computational Bioscience Research Center, King Abdullah University of Science and Technology</li>
</ul>

<h1>
<a id="tutorial-outline" class="anchor" href="#tutorial-outline" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Tutorial outline</h1>

<table>
<thead>
<tr>
<th><strong>Time</strong></th>
<th align="right"><strong>Topic</strong></th>
<th><strong>Presenter</strong></th>
<th><strong>Slides</strong></th>
<th align="right"><strong>Lecture materials, additional reading</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>9:00-9:45</td>
<td align="right">Welcome and introduction; the role of ontologies in biology, biomedicine and disease genetics</td>
<td>Paul Schofield</td>
<td>tbd</td>
<td align="right"><a href="http://bib.oxfordjournals.org/content/16/6/1069">The role of ontologies in biological and biomedical research: a functional perspective</a></td>
</tr>
<tr>
<td>9:45-11:00</td>
<td align="right">Ontology Basics: classes, relations, axioms; relations between ontologies and graphs; the Web Ontology Language (OWL)</td>
<td>Robert Hoehndorf</td>
<td>tbd</td>
<td align="right"><a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-441">Relations as patterns: bridging the gap between OBO and OWL</a></td>
</tr>
<tr>
<td>10:30-11:00</td>
<td align="right">Coffee Break</td>
<td>all</td>
<td></td>
<td align="right"></td>
</tr>
<tr>
<td>11:00-12:30</td>
<td align="right">Phenotype ontologies: PATO and EQ</td>
<td>George Gkoutos</td>
<td>tbd</td>
<td align="right"><a href="http://www.ncbi.nlm.nih.gov/pubmed/14992502">Building mouse phenotype ontologies</a></td>
</tr>
<tr>
<td>12:30-13:30</td>
<td align="right">Lunch break</td>
<td>All</td>
<td></td>
<td align="right"><a href="http://www.webmd.com/diet/">Diet and Weight Management</a></td>
</tr>
<tr>
<td>13:30-14:15</td>
<td align="right">Informatics of phenotype ontologies, computational analysis of phenotype data</td>
<td>Robert Hoehndorf</td>
<td><a href="https://github.com/bio-ontology-research-group/ontology-and-phenotype-tutorial/raw/master/informatics-of-phenotypes/informatics.pdf">Lecture slides</a></td>
<td align="right"><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3185433/">PhenomeNET: a whole-phenome approach to disease gene discovery</a></td>
</tr>
<tr>
<td>14:15-15:00</td>
<td align="right">The role of ontologies in the analysis of personal genome data</td>
<td>Imene Boudellioua</td>
<td>tbd</td>
<td align="right"><a href="http://genome.cshlp.org/content/24/2/340.long">Improved exome prioritization of disease genes through cross-species phenotype comparison</a></td>
</tr>
<tr>
<td>15:00-15:30</td>
<td align="right">Coffee Break</td>
<td>all</td>
<td></td>
<td align="right"></td>
</tr>
<tr>
<td>15:30-16:15</td>
<td align="right">Complex diseases, physiology, disease modules and networks</td>
<td>Paul Schofield</td>
<td>tbd</td>
<td align="right"></td>
</tr>
<tr>
<td>16:15-17:00</td>
<td align="right">Discussion: discuss problems brought up by the audience; ontologies and the EHR; ontology mapping</td>
<td>all</td>
<td>tbd</td>
<td align="right"></td>
</tr>
</tbody>
</table>

<h1>
<a id="participant-requirements" class="anchor" href="#participant-requirements" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Participant Requirements</h1>

<p>All participants must have a laptop with Protege installed and a web browser to access BioPortal and AberOWL.</p>

<h1>
<a id="materials" class="anchor" href="#materials" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Materials</h1>

<p>Here you will soon find all materials developed for the tutorial, including slides, lecture notes, software tools and reading material.</p>

<h2>
<a id="slides" class="anchor" href="#slides" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Slides</h2>

<h2>
<a id="lecture-notes" class="anchor" href="#lecture-notes" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Lecture notes</h2>

<h2>
<a id="worksheets" class="anchor" href="#worksheets" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Worksheets</h2>

<p>Worksheets contain small problems to be solved in the tutorial.</p>

<ol>
<li>Finding and browsing an ontology</li>
<li>Editing an ontology in Protege: understanding classes, relations and axioms</li>
<li>Annotating with the Phenotype and Trait Ontology (PATO) and the Entity-Quality (EQ) formalism.</li>
<li>Generating a phenotype ontology from EQ annotations</li>
<li>Finding causal variants in personal genome sequences: analyzing a Variant Call Format (VCF) file</li>
</ol>

<h2>
<a id="software-and-libraries" class="anchor" href="#software-and-libraries" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Software and libraries</h2>

<p>Working with ontologies:</p>

<ul>
<li><a href="http://protege.stanford.edu/">Protege Ontology Editor</a></li>
<li><a href="http://bioportal.bioontology.org">BioPortal</a></li>
<li><a href="http://aber-owl.net">AberOWL</a></li>
</ul>

<p>Ontology Matching:</p>

<ul>
<li><a href="https://github.com/AgreementMakerLight">AgreementMaker Light</a></li>
</ul>

<p>Analyzing genomic data with ontologies:</p>

<ul>
<li><a href="http://www.sanger.ac.uk/science/tools/exomiser">Exomiser</a></li>
<li><a href="http://phenomebrowser.net">PhenomeNET</a></li>
</ul>

<p>Libraries, mainly for developers (optional, not required for the tutorial):</p>

<ul>
<li><a href="http://owlapi.sourceforge.net">OWL API</a></li>
<li><a href="https://github.com/liveontologies/elk-reasoner">ELK Reasoner</a></li>
<li><a href="http://www.semantic-measures-library.org/">Semantic Measures Library</a></li>
</ul>

<h2>
<a id="additional-reading-material" class="anchor" href="#additional-reading-material" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Additional reading material</h2>

<h1>
<a id="contact" class="anchor" href="#contact" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Contact</h1>

<p>For any questions related to the tutorial, or requests for topics to cover in the course, please send a mail to <a href="mailto:phenotype-ontology-tutorial@googlegroups.com">phenotype-ontology-tutorial@googlegroups.com</a>. The mailinglist will remain active after the tutorial and can be used for post-tutorial discussions.</p>
        </section>

        <footer>
          Bio-ontologies and their role in analyzing personal genome data is maintained by <a href="https://github.com/bio-ontology-research-group">bio-ontology-research-group</a><br>
          This page was generated by <a href="https://pages.github.com">GitHub Pages</a>. Tactile theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.
        </footer>

        
      </div>
    </div>
  </body>
</html>
